composer/utils/iter_helpers.py

```diff
@@ -8,6 +8,21 @@
 import contextlib
 
 
+def recursive_apply(func: callable, data):
```

Comment:
is there a way to merge this and/or use `map_collection` below? Syntactically they look similar. If not, at minimum they should match signature -- e.g. data, func to match the collection, map_fn below.

Reply:
Map_collection isn't recursive, it's pretty a switch statement between dict, lists, and single items. 

I could rename recursive apply to map_collection possibly? The functionality should be a superset of the existing function.

Reply:
The function signature will also need type annotations in `iter_helpers.pyi`.